File: Reverse_Only_Letters/Reverse_only_letters.java
public class Main {
    public static void main(String[] args) {
        process("ab-cd");
        process("a-bC-dEf-ghIj");
        process("Test1ng-Leet=code-Q!");
    }
    private static void process(String s) {
        char[] charArray = s.toCharArray();
        int i = 0;
        int j = charArray.length - 1;
        while (i < j) {
            if (Character.isLetter(charArray[i]) && Character.isLetter(charArray[j])) {
                char temp = charArray[i];
                charArray[i] = charArray[j];
                charArray[j] = temp;
                i++;
                j--;
            } else if (!Character.isLetter(charArray[j])) {
                j--;
            } else {
                i++;
            }
        }
        String result = new String(charArray);
        System.out.println(result);
    }
}

File: Reverse_Only_Letters/Reverse_only_letters.py
def reverse_only_letters(s):
    s = list(s)
    i = 0
    j = len(s)-1

    while i < j:
        if s[i].isalpha() and s[j].isalpha():
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1
        elif s[i].isalpha() and not s[j].isalpha():
            j -= 1
        else:
            i += 1

    return "".join(s)

print(reverse_only_letters(s="ab1-cd"))
print(reverse_only_letters(s="a-bC-dEf-ghIj"))
print(reverse_only_letters(s="Test1ng-Leet=code-Q!"))

